File: testdata/fixtures/ts-dirty/src/domain/order.ts
// Domain layer: should stay pure (std only) — but here it reaches UP into the
// service layer, a layering inversion the rules forbid.
import { randomUUID } from "node:crypto";

import { notify } from "../service/notify";

export interface Order {
  id: string;
  total: number;
}

export function newOrder(total: number): Order {
  const order = { id: randomUUID(), total };
  notify(order.id);
  return order;
}

File: testdata/fixtures/kotlin-dirty/src/main/kotlin/com/example/domain/Order.kt
// Domain layer: should stay pure (std only) — but here it reaches UP into the
// service.notify layer, a layering inversion the rules forbid.
package com.example.domain

import java.util.concurrent.atomic.AtomicLong

import com.example.service.notify.Notifier

class Order(val total: Int) {
    val id: Long = NEXT_ID.getAndIncrement()

    init {
        Notifier.notify(id)
    }

    override fun toString(): String = "order $id ($total)"

    companion object {
        private val NEXT_ID = AtomicLong(1)
    }
}
